source/ada/lsp-ada_documents.adb

```diff
@@ -604,6 +605,314 @@ package body LSP.Ada_Documents is
+
+         procedure Prepare (Insert : Boolean);
+         procedure Prepare
+           (Insert : Boolean;
+            Char   : VSS.Characters.Virtual_Character);
+         --  Store information for Text_Edit in Changed_Block_Text and
+         --  Changed_Block_Span
+
+         procedure Add_Prepared;
+         --  Add prepared New_String and Span into Text_Edit
+
+         procedure Backward is
+         begin
+            if not Cursor.Backward
+              and then Old_Lines_Number > 1
+            then
+               Current_Line_Number := Current_Line_Number - 1;
+               Old_Lines_Number    := Old_Lines_Number - 1;
+               Old_Line            := Old_Lines.Element (Old_Lines_Number);
+               Cursor.Set_At_Last (Old_Line);
+            end if;
+
+            Old_Index := Old_Index - 1;
+            Dummy     := Old_Char.Backward;
+         end Backward;
+
+         ------------------
+         -- Get_Position --
+         ------------------
+
+         function Get_Position (Insert : Boolean) return Position
+         is
+            --------------
+            -- Backward --
+            --------------
+
+            function Backward return Position;
+            function Backward return Position is
+               C : VSS.Strings.Cursors.Iterators.Characters.
+                 Character_Iterator := Old_Line.At_Character (Cursor);
+            begin
+               --  "Cursor" is after the current character but we should
+               --  inserting before it
+               if C.Backward then
+                  return
+                    (line      => Current_Line_Number,
+                     character => C.First_UTF16_Offset);
+               else
+                  return
+                    (line      => Current_Line_Number,
+                     character => 0);
+               end if;
+            end Backward;
+
+         begin
+            if not Cursor.Has_Element then
+               return
+                 (line      => Current_Line_Number,
+                  character => 0);
+
+            elsif Insert then
+               --  "Cursor" is after the current character but we should
+               --  inserting before it
+               return Backward;
+
+            else
+               return
+                 (line      => Current_Line_Number,
+                  character => Cursor.First_UTF16_Offset);
+            end if;
+         end Get_Position;
+
+         -------------
+         -- Prepare --
+         -------------
+
+         procedure Prepare (Insert : Boolean) is
+         begin
+            if not Span_Set then
+               --  it is the first portion of a changed block so store
+               --  last position of the changes
+               Span_Set := True;
+               Changed_Block_Span.last := Get_Position (Insert);
+            end if;
+         end Prepare;
+
+         -------------
+         -- Prepare --
+         -------------
+
+         procedure Prepare
```

Comment:
Can't we have a more precise name than 'Prepare'?

Reply:
Changed